### ISSUE TITLE
Allow cache exceptions aka. "white exceptions"

### ISSUE THREAD

**Issue by Rebzzel** [Krukov/cashews#143]
Hi, @Krukov, nice library!
I came up with a possible feature.

Imagine that we have a function about which we know that if an exception occurs in it, then this exception will definitely be thrown for the next 5 minutes.

So we want this function (after `cache') to throw exceptions immediately without calling its body.

I imagine the api working something like this:
```py
@cache(ttl="5m", white_exceptions=(MyException,))
async def my_func(a, b):
    print("called!")
    ...
    raise MyException(...)


# first call
await my_func(1, 2)
# called!
# ... [traceback] ...

...

# after 1 sec
await my_func(1, 2)
# ... [traceback] ...

...

# after 5 mins
await my_func(1, 2)
# called!
# ... [traceback] ...
```

Here is my hack to get this work on current version:

```py
class WhiteExceptionContainer:
    def __init__(self, exception: Exception):
        self.exception = exception
        
        
def with_white_exceptions(cache_decorator):
    @wraps(cache_decorator)
    def wrap(
        *cache_decorator_args,
        white_exceptions: typing.Iterable[Exception] = tuple(),
        **cache_decorator_kwargs,
    ):
        def decorator(func):
            @wraps(func)
            async def execute_raw(*args, **kwargs):
                try:
                    return await func(*args, **kwargs)
                except white_exceptions as exception:
                    return WhiteExceptionContainer(exception)

            func_cached = cache_decorator(
                *cache_decorator_args, **cache_decorator_kwargs
            )(execute_raw)

            @wraps(func)
            async def execute_cached(*args, **kwargs):
                result = await func_cached(*args, **kwargs)

                if isinstance(result, WhiteExceptionContainer):
                    raise result.exception

                return result
            return execute_cached
        return decorator
    return wrap


cache = with_white_exceptions(cache)


@cache(ttl="5m", white_exceptions=(MyException,))
async def my_func(a, b):
    ...
```

**Comment by Krukov** [comment]
Hi, 
Thanks, I got your idea and I thought about this case a few month ago and decided that in most cases it will be better just replace exception with returning some "meaningful" value  e.g. `None` - but now I do not like this solution, because of  changing a function signature just for caching, so I agree with you that it will be better just to cache an exception if user want it.
One point I have - I prefer to use  existing argument `condition` which is responsible for determining when to cache or not ( I just don't want to add a new argument because there are so many of them). What do you think? 
```python
@cache(ttl="5m", condition=with_exceptions(MyException))
async def my_func(a, b):
    print("called!")
    ...
    raise MyException(...)
```

**Comment by Rebzzel** [comment]
> will be better just replace exception with returning some "meaningful" value e.g. None - but now I do not like this solution, because of changing a function signature just for caching

In addition:

1. Sometimes useful information is stored in exceptions, so just `None` loses this.
2. This also breaks the logic of `try: ... catch: ...`.

> I prefer to use existing argument `condition`

In this case, I suggest pass exception into `result`:

```py
def my_condition(result: Any | Exception, args, kwargs, key=None):
    if isinstance(result, Exception):
        # in this case cachews must raise exception after call condition
        if instance(result, MyException):
            return True
        else:
            return False 
    return True
```

But start using `condition` for this may be breaking change 🤔

**Comment by Rebzzel** [comment]
Sorry, I was inactive for a while.

To be honest, it doesn't look very intuitive, but I don't have any better ideas than that (if using `condition`).

**Comment by Krukov** [comment]
just released a new version (6.2.0) with those condition - plz check this 

> It is also possible to cache an exception that function can raise, to do so use special conditions (only for simple, hit and early cache)

```python
from cashews import cache, with_exceptions, only_exceptions

cache.setup("mem://")

@cache(ttl="1h", condition=with_exceptions(MyException, TimeoutError))
async def get():
    ...


@cache(ttl="1h", condition=only_exceptions(MyException, TimeoutError))
async def get():
    ...
```

**Comment by Rebzzel** [comment]
This just works. Good job and thank you!
